chain/lf3/f3.go

```diff
@@ -56,32 +60,119 @@ type F3 struct {
 type F3Params struct {
 	fx.In
 
-	PubSub       *pubsub.PubSub
-	Host         host.Host
-	ChainStore   *store.ChainStore
-	Syncer       *chain.Syncer
-	StateManager *stmgr.StateManager
-	Datastore    dtypes.MetadataDS
-	Wallet       api.Wallet
-	Config       *Config
-	LockedRepo   repo.LockedRepo
+	PubSub        *pubsub.PubSub
+	Host          host.Host
+	ChainStore    *store.ChainStore
+	Syncer        *chain.Syncer
+	StateManager  *stmgr.StateManager
+	MetaDatastore dtypes.MetadataDS
+	F3Datastore   dtypes.F3DS
+	Wallet        api.Wallet
+	Config        *Config
+	LockedRepo    repo.LockedRepo
 
 	Net api.Net
 }
 
 var log = logging.Logger("f3")
 
+var migrationKey = datastore.NewKey("/migration")
```

Comment:
Use a key with a less chance of accidental conflict with a pre-existing one? e.g. add `f3` somewhere in there?
```suggestion
var migrationKey = datastore.NewKey("/f3-migration/1")
```
```suggestion
var migrationKey = datastore.NewKey("/migration")
```